packages/typespec-ts/src/utils/clientUtils.ts

```diff
@@ -3,17 +3,31 @@ import {
   SdkClient,
   SdkClientType,
   SdkServiceOperation,
-  listAllServiceNamespaces
+  listAllServiceNamespaces,
+  listClients
 } from "@azure-tools/typespec-client-generator-core";
-import { getNamespaceFullName } from "@typespec/compiler";
+import {
+  getNamespaceFullName,
+  Interface,
+  isTemplateDeclaration,
+  isTemplateDeclarationOrInstance,
+  Namespace,
+  Operation
+} from "@typespec/compiler";
 import { SdkContext } from "./interfaces.js";
 import { ModularClientOptions } from "../modular/interfaces.js";
 import { NameType, normalizeName } from "@azure-tools/rlc-common";
 
 export function getRLCClients(dpgContext: SdkContext): SdkClient[] {
-  const services = listAllServiceNamespaces(dpgContext);
+  const services = new Set<Namespace>();
+  listClients(dpgContext).map((c) => services.add(c.service));
+  const rawServiceNamespaces = listAllServiceNamespaces(dpgContext);
```

Comment:
where is the logic to filter out the multiple namespaces but client.tsp only import part of them?

Reply:
We only calculate the service which in clients by `listClients(dpgContext)` so all services are in client's scopes.